marshal_test.go

```diff
@@ -368,3 +399,22 @@ type failJSONMarshaler string
 func (*failJSONMarshaler) MarshalJSON() ([]byte, error) {
 	return nil, errgo.New("marshal error")
 }
+
+func (*marshalSuite) TestMarshalHeaderSetter(c *gc.C) {
+	req, err := httprequest.Marshal("http://example.com", "GET", &headerSetterTester{
+		A: "a",
+		B: "b",
+	})
+	c.Assert(err, gc.IsNil)
+	c.Assert(req.Header["A"], gc.DeepEquals, []string{"a"})
+	c.Assert(req.Header["B"], gc.DeepEquals, []string{"b"})
+}
```

Comment:
I'd also like to see a test using CustomHeader on a marshal body.
(it won't work, but it would be nice to either clarify that it won't work
in CustomHeader itself or explicitly implement support for CustomHeader in Marshal)